src/frontend/js/pages/TeacherDashboardContractsLayout/components/BulkDownloadContractButton/index.tsx

```diff
@@ -0,0 +1,75 @@
+import { Button } from '@openfun/cunningham-react';
+import { FormattedMessage, defineMessages } from 'react-intl';
+import { useEffect } from 'react';
+import useDownloadContractArchive, {
+  ContractDownloadStatus,
+} from 'pages/TeacherDashboardContractsLayout/hooks/useDownloadContractArchive';
+import { Organization } from 'types/Joanie';
+
+const messages = defineMessages({
+  bulkDownloadButtonDownloadLabel: {
+    defaultMessage: 'Download fully signed contracts',
+    description: 'The label of the bulk download button when the zip archive is ready for download',
+    id: 'pages.TeacherDashboardContractsLayout.BulkDownloadContractButton.bulkDownloadButtonDownloadLabel',
+  },
+  bulkDownloadButtonPendingLabel: {
+    defaultMessage: 'Waiting zip archive generation...',
+    description: 'The label of the bulk download button when archive generation is pending',
+    id: 'pages.TeacherDashboardContractsLayout.BulkDownloadContractButton.bulkDownloadButtonPendingLabel',
+  },
+  bulkDownloadButtonRequestArchiveLabel: {
+    defaultMessage: 'Request zip archive generation',
+    description: 'The label of the bulk download button to request the generation of a zip archive',
+    id: 'pages.TeacherDashboardContractsLayout.BulkDownloadContractButton.bulkDownloadButtonRequestArchiveLabel',
+  },
+});
+
+interface BulkDownloadContractButtonProps {
+  organizationId: Organization['id'];
+}
+
+const BulkDownloadContractButton = ({ organizationId }: BulkDownloadContractButtonProps) => {
+  const { doDownloadContractArchive, doCreateContractArchive, status } = useDownloadContractArchive(
+    {
+      organizationId,
+    },
+  );
+
+  useEffect(() => {
+    // Trigger contract's archive polling when generation had already been requested
+    if (status === ContractDownloadStatus.PENDING) {
+      doCreateContractArchive();
```

Comment:
Side note: related to this kind of inverted dependency ( in my opinion ), I added some `console.log`, and here is the result when we click on the IDLE button:
<img width="583" alt="Capture d’écran 2024-01-23 à 12 11 01" src="https://github.com/openfun/richie/assets/9628870/d2166a3d-f61b-4dac-9f08-118820ade579">

<img width="511" alt="Capture d’écran 2024-01-23 à 12 13 03" src="https://github.com/openfun/richie/assets/9628870/b62fe3e0-8ed5-450e-b8e5-680480295477">

<img width="323" alt="Capture d’écran 2024-01-23 à 12 11 33" src="https://github.com/openfun/richie/assets/9628870/da6a88fc-6c13-460d-8f2d-aa704cffc312">


I know that your code is protected against that but we can see that `doCreateContractArchive` is called two times in a row. We must keep this in mind in order to avoid any side effect in the future if we want to evolve this feature.

Reply:
I could add polling state in useContractDownload hook to avoid this double call.